freqtrade/optimize/backtesting.py

```diff
@@ -350,8 +350,35 @@ def _get_close_rate(self, sell_row: Tuple, trade: LocalTrade, sell: SellCheckTup
         else:
             return sell_row[OPEN_IDX]
 
+    def _get_adjust_trade_entry_for_candle(self, trade: LocalTrade, row: Tuple
+                                           ) -> LocalTrade:
+
+        # If there is any open orders, wait for them to finish.
+        for order in trade.orders:
+            if order.ft_is_open:
+                return trade
+
+        current_profit = trade.calc_profit_ratio(row[OPEN_IDX])
+        stake_amount = strategy_safe_wrapper(self.strategy.adjust_trade_position,
+                                             default_retval=None)(
+            pair=trade.pair, trade=trade, current_time=row[DATE_IDX].to_pydatetime(),
+            current_rate=row[OPEN_IDX], current_profit=current_profit)
+
+        # Check if we should increase our position
+        if stake_amount is not None and stake_amount > 0.0:
+            pos_trade = self._enter_trade(trade.pair, row, stake_amount, trade)
+            if pos_trade is not None:
+                return pos_trade
```

Comment:
I think you can omit these 2 lines here by reassigning `trade` (which would not even be necessary - but i'd keep for clarity, as non-primitives (int, float, ...) are passed by reference - so modifications to the trade object from within the method will always affect the original trade object, too.)

Reply:
It actually initially was a single line, but _enter_trade is capable of returning None in case there's not enough wallet etc and as such its return value is Optional[Trade] and this brings mypy warnings.